test/model/content_test.dart

```diff
@@ -521,6 +521,118 @@ class ContentExample {
       blockUnimplemented('more text'),
     ]]),
   ]);
+
+  static const videoEmbedYoutube = ContentExample(
+    'video preview for youtube embed with thumbnail',
+    null,
+    '<p>'
+      '<a href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">https://www.youtube.com/watch?v=aqz-KE-bpKQ</a>'
+    '</p>\n'
+    '<div class="youtube-video message_inline_image">'
+      '<a data-id="aqz-KE-bpKQ" href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">'
+        '<img src="/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"></a></div>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://www.youtube.com/watch?v=aqz-KE-bpKQ', nodes: [TextNode('https://www.youtube.com/watch?v=aqz-KE-bpKQ')]),
+    ]),
+    VideoNode(
+      srcUrl: "https://www.youtube.com/watch?v=aqz-KE-bpKQ",
+      previewImageUrl: "/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"
+    ),
+  ]);
+
+  static const videoEmbedYoutubeClassNameReordered = ContentExample(
+    'video preview for youtube embed with thumbnail, (hypothetical) class name reorder',
+    null,
+    '<p>'
+      '<a href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">https://www.youtube.com/watch?v=aqz-KE-bpKQ</a>'
+    '</p>\n'
+    '<div class="message_inline_image youtube-video">'
+      '<a data-id="aqz-KE-bpKQ" href="https://www.youtube.com/watch?v=aqz-KE-bpKQ">'
+        '<img src="/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"></a></div>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://www.youtube.com/watch?v=aqz-KE-bpKQ', nodes: [TextNode('https://www.youtube.com/watch?v=aqz-KE-bpKQ')]),
+    ]),
+    VideoNode(
+      srcUrl: "https://www.youtube.com/watch?v=aqz-KE-bpKQ",
+      previewImageUrl: "/external_content/ecb96e8f884f481c4bc0179287a44ab9014aa78f/68747470733a2f2f692e7974696d672e636f6d2f76692f61717a2d4b452d62704b512f64656661756c742e6a7067"
+    ),
+  ]);
+
+  static const videoEmbedVimeoPreviewDisabled = ContentExample(
+    'video preview for vimeo embed with realm link previews disabled',
+    null,
+    '<p>'
+      '<a href="https://vimeo.com/1084537">https://vimeo.com/1084537</a></p>', [
+    ParagraphNode(links: null, nodes: [
+      LinkNode(url: 'https://vimeo.com/1084537', nodes: [TextNode('https://vimeo.com/1084537')]),
+    ]),
+  ]);
+
+  static const videoEmbedVimeoPreviewEnabled = ContentExample(
```

Comment:
```suggestion
  static const videoEmbedVimeo = ContentExample(
```

Given that the preview-disabled case isn't really a "video embed" at all, we can treat this as the main Vimeo example and give it a shorter name.